crates/sparrow-runtime/src/execute/key_hash_inverse.rs

```diff
@@ -105,7 +105,7 @@ impl KeyHashInverse {
     /// values are aligned to map from a key to a hash per index. The
     /// current implementation eagerly adds the keys and hashes to the
     /// inverse but can be optimized to perform the addition lazily.
-    fn add(&mut self, keys: ArrayRef, key_hashes: &UInt64Array) -> anyhow::Result<()> {
+    pub fn add(&mut self, keys: ArrayRef, key_hashes: &UInt64Array) -> anyhow::Result<()> {
```

Comment:
Does this need to be public? I believe that we had this hidden because users should interact with this via the "thread-safe key hash inverse" (otherwise it's not... thread safe...)

Given the protections on mutability, it's probably *ok* to have this be public... just wanted to make sure it was necessary.

Reply:
You are correct - reverted this change.